File: KULIAH/Praktikum1/app/src/main/java/id/ac/unsyiah/android/praktikum1/MainActivity.java
package id.ac.unsyiah.android.praktikum1;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    Button button;
    EditText edit_nama;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        button=findViewById(R.id.button2);
        edit_nama = findViewById(R.id.edit_nama);
   /*
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String name = edit_nama.getText().toString();
                Toast.makeText(MainActivity.this, "Selamat Datang", Toast.LENGTH_SHORT).show();
            }
        });
    */
    }

    public void tekanTombol(View view) {
        String name = edit_nama.getText().toString();
        Toast.makeText(MainActivity.this, "Selamat Datang", Toast.LENGTH_SHORT).show();
    }
}

File: KULIAH/PhotoApp/app/src/main/java/id/ac/unsyiah/android/photoapp/MainActivity.kt
package id.ac.unsyiah.android.photoapp

import android.os.Bundle
import android.support.design.widget.FloatingActionButton
import android.support.design.widget.Snackbar
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.support.v7.widget.Toolbar
import android.util.Log
import android.view.View
import android.view.Menu
import android.view.MenuItem
import id.ac.unsyiah.android.photoapp.api.PhotoRetriever
import id.ac.unsyiah.android.photoapp.models.Photo
import id.ac.unsyiah.android.photoapp.models.PhotoList
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity(), View.OnClickListener {
    override fun onClick(v: View?) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    var photos: List<Photo>? = null
    var mainAdapter : MainAdapter? =null
    lateinit var recyclerView: RecyclerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        recyclerView = findViewById(R.id.recyclerView) as RecyclerView
        recyclerView.layoutManager = LinearLayoutManager(this)

        var retriever = PhotoRetriever()

        var callback = object : Callback<PhotoList> {
            override fun onFailure(call: Call<PhotoList>, t: Throwable) {
                Log.e("MainActivity", "Gagal memanggil API")
            }

            override fun onResponse(call: Call<PhotoList>, response: Response<PhotoList>) {
                response?.isSuccessful.let {
                    this@MainActivity.photos = response?.body()?.hits
                    mainAdapter = MainAdapter(this@MainActivity.photos!!, this@MainActivity)
                    recyclerView.adapter = mainAdapter
                }
            }

        }

        retriever.getPhotos(callback)

    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        val id = item.itemId


        return if (id == R.id.action_settings) {
            true
        } else super.onOptionsItemSelected(item)

    }
}
